crates/bitcoin-da/tests/bitcoin_da_tests.rs

```diff
@@ -357,40 +409,220 @@ mod tests {
+
+        // Set up embedded data and relayer configuration
         let embedded_data = b"Hello, world!";
         let relayer = Relayer::new(&Config::new(
-            // "YOUR_NODE_IP:38332".to_owned(), // SIGNET
-            "localhost:8332".to_owned(), // REGNET
+            NODE_IP.to_owned(),
             "rpcuser".to_owned(),
             "rpcpass".to_owned(),
         ))
         .unwrap();
+
         // get network, should be regtest
         let blockchain_info = relayer.client.get_blockchain_info().unwrap();
         let network_name = &blockchain_info.chain;
-        let network = Network::from_core_arg(network_name)
+        let _network = Network::from_core_arg(network_name)
             .map_err(|_| BitcoinError::InvalidNetwork)
             .unwrap();
-        // assert_eq!(network, Network::Regtest);
-        assert_eq!(network, Network::Signet);
 
+        #[cfg(feature = "signet")]
+        assert_eq!(_network, Network::Signet);
+
+        #[cfg(feature = "regtest")]
+        assert_eq!(_network, Network::Regtest);
+
+        // ==================================
+        // When: data is written to the network
+        // ==================================
+
+        let write_result = relayer.write(embedded_data);
+
+        // ==========================================
+        // Then: assert successful writing operation
+        // ==========================================
+
+        match write_result {
+            Ok(txid) => {
+                println!("Txid: {}", txid);
+                println!("Successful write");
+            }
+            Err(e) => panic!("Write failed with error: {:?}", e),
+        }
+    }
+
+    #[cfg(feature = "long_tests")]
+    #[test]
+    fn test_read_data_by_block_height() {
+        // ===============================
+        // Given: a block with embedded data
+        // ===============================
+    
+        // Prepare the data and relayer configuration
+        let embedded_data = b"Hello, height!";
+        let relayer = Relayer::new(&Config::new(
+            NODE_IP.to_owned(),
+            "rpcuser".to_owned(),
+            "rpcpass".to_owned(),
+        ))
+        .unwrap();
+    
+        // Obtain the current block height before embedding
+        let blockchain_info = relayer.client.get_blockchain_info().unwrap();
+        let current_height = blockchain_info.blocks;
+    
+        // Embed the data into the blockchain by writing a transaction
         match relayer.write(embedded_data) {
             Ok(txid) => {
                 println!("Txid: {}", txid);
                 println!("Successful write");
             }
             Err(e) => panic!("Write failed with error: {:?}", e),
         }
+    
+        // Add the transaction to a new block
+        relayer.generate_blocks(1).unwrap();
+    
+        if cfg!(feature = "regtest") {
+            // Generate a block immediately in regtest
+            relayer.generate_blocks(1).unwrap();
```

Comment:
Is there a rational to do it twice ?

Reply:
Not really. I made a mistake. Removed and tested. Have not pushed yet.

Reply:
pushed